### DISCUSSION TITLE
Python: "No Python interpreter configured for this project"

### DISCUSSION THREAD

**Discussion by noamraph** [JetBrains/Qodana#61] [Q&A]
Hi,

I tried to analyze my Python project using the jetbrains/qodana-python docker image.
The most common error is "An invalid interpreter", I think that all files have it.

I can ignore those errors, perhaps that's the immediate solution. I hope it would be possible to specify an interpreter, because I believe that many inspections actually rely on an interpreter.

Thanks,
Noam

**Comment by brichbash** [comment]
Hi,

In case if you have `venv` in your project, and you didn’t create it using Qodana image, Qodana cannot resolve the interpreter as `venv` contains only symlinks to it. You can see more detailed info in [our docs](https://www.jetbrains.com/help/qodana/qodana-python.html#Try+it+now) that explains how to install dependencies at the moment.
>I hope it would be possible to specify an interpreter

You can specify an interpreter installing it via `pyenv` in the container. Please see the following samples:
First run to install all dependencies and the required interpreter:
```
docker run --rm \
-v <source-directory>/:/data/project/ \
-v <cache-directory>/:/data/cache \
-v <cache-directory>/py/:/opt/venv/versions/ \
--entrypoint=bash \
jetbrains/qodana-python -c '
pyenv install <python-version>
pyenv global <python-version>
python3 -m venv /data/cache/venv
source /data/cache/venv/bin/activate
pip3 install -r /data/project/requirements.txt'
```
Second run for the analysis:
```
docker run --rm -it -p 8080:8080 \
-v $(pwd)/:/data/project/ \
-v <cache-directory>/:/data/cache/ \
-v $(pwd)/res:/data/results/ \
-v <cache-directory>/py/:/opt/pyenv/versions/ \
jetbrains/qodana-python --show-report
```
After the first run dependencies and required python interpreter are stored in your cache directory, and when running Qodana the next time it will pick up all dependencies from the cache.
